### PR TITLE
it seems that pushd and popd may be too smart for their own good

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
It seems that pushd and popd are causing issues, note that despite `os.chdir` being used to move into the tmp dir the file is still saved in the local dir. Changing this over to `cd` seems to help.

**Comment by CJ-Wright** [comment]
Attn:
@scopatz @gforsyth

**Comment by gforsyth** [comment]
Hey @CJ-Wright -- there seem to be a bunch of test failures here

**Comment by CJ-Wright** [comment]
Yes I think that the chdir is leaving it in the wrong dir. I'll try to fix it soon (I'll write a python indir)

**Comment by scopatz** [comment]
Let us know when this is ready for another look!

**Comment by CJ-Wright** [comment]
Cycling for ci

**Comment by CJ-Wright** [comment]
@gforsyth @scopatz I'm not certain why the tests are failing on win/osx. Any ideas?

**Comment by gforsyth** [comment]
Travis has spurious failures, just restarted that job.  Let me check out Appveyor now.

**Comment by gforsyth** [comment]
I think Appveyor is also failing in a spurious way, but I cannot figure out how to restart the job which is crazy making.

**Comment by gforsyth** [comment]
Honestly, don't even worry about it.  The failure on Appveyor is unrelated, so I think we can call this a "pass", so if @scopatz is good on the latest changes this can go in.

File: xonsh/lib/os.xsh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """Xonsh extension of the standard library os module, using xonsh for
 subprocess calls"""
 from contextlib import contextmanager
+import os
```
Comment (CJ-Wright) [inline xonsh/lib/os.xsh]: this is not needed

**Comment by scopatz** [comment]
Yeah, restarting appveyor is insane

**Comment by scopatz** [comment]
Thanks @CJ-Wright

**Pull request merged**